### PR TITLE
Added an additional stream control

### PR DESCRIPTION
**Pull request by ravil-mobile**

### CODE REVIEW

File: interfaces/common/Common.h [hunk 1]
```diff
@@ -26,4 +29,19 @@ U align(T number, U alignment) {
 }
 }
 
-#endif
\ No newline at end of file
+inline int getProcessId() {
+  std::vector<std::string> rankEnvVars{{"SLURM_PROCID"},
```
Comment (krenzland) [inline interfaces/common/Common.h:33]: what exactly is the difference to the MPI rank?

File: interfaces/cuda/Control.cu [hunk 2]
```diff
@@ -147,6 +147,13 @@ void ConcreteAPI::finalize() {
   if (status[StatusID::InterfaceInitialized]) {
     cudaStreamDestroy(defaultStream); CHECK_ERR;
     cudaEventDestroy(defaultStreamEvent); CHECK_ERR;
+    if (not genericStreams.empty()) {
```
Comment (krenzland) [inline interfaces/cuda/Control.cu]: I still believe that not is a very strange coding style :(

File: interfaces/hip/Control.cpp [hunk 3]
```diff
@@ -138,6 +138,13 @@ void ConcreteAPI::finalize() {
   if (status[StatusID::InterfaceInitialized]) {
     hipStreamDestroy(defaultStream); CHECK_ERR;
     hipEventDestroy(defaultStreamEvent); CHECK_ERR;
+    if (not genericStreams.empty()) {
```
Comment (krenzland) [inline interfaces/hip/Control.cpp]: !

**Review by krenzland — CHANGES_REQUESTED** [review]
I'm slightly confused by getProcessId - most things are more comments than suggestions though

File: interfaces/common/Common.h (hunk 1 above)
Comment (ravil-mobile) [reply interfaces/common/Common.h:33]: renamed to `getMpiRankFromEnv`. Note, we need MPI rank only for outputting some debug/info to the console.

The device instance should be initialized before a call to MPI_Init (it used to be a requirement for some GPU-Aware MPI implementation). The safest option would be reading the rank from the env. variable.

File: interfaces/cuda/Control.cu (hunk 2 above)
Comment (ravil-mobile) [reply interfaces/cuda/Control.cu]: reworked. Now, it is `!genericStreams.empty()`

PS I don't know. It looks more informative/readable for me. The only one thing to keep in mind that `not` works only with `bool`. But `empty()` always returns bool. However, I just changed as you requested.

File: interfaces/hip/Control.cpp (hunk 3 above)
Comment (ravil-mobile) [reply interfaces/hip/Control.cpp]: done

**Review by krenzland — APPROVED** [review]

**Pull request merged**